flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/StateChangelogStorageFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.changelog;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.configuration.Configuration;
+
+/**
+ * A factory for {@link StateChangelogStorage}. Please use {@link StateChangelogStorageLoader} to
+ * create {@link StateChangelogStorage}.
+ */
+@Internal
```

Comment:
I agree that users might eventually want to extend it, but I'm not sure that it is in int's final shape. So I'd keep it internal for one or two releases to avoid changing public (or even `PublicEvloving`) API. 
WDYT?

Reply:
I second @rkhachatryan's opinion. Maybe we reconsider this later.

Reply:
Okay, maybe we could add some TODO here to remind developers.

Reply:
@Myasuka I'll create a ticket to track this.